src/Core/Framework/Test/Api/Serializer/JsonApiEncoderTest.php

```diff
@@ -140,6 +203,101 @@ public function testEncodeStructWithToManyExtension(): void
         $this->assertValues($fixture->getAdminJsonApiFixtures(), json_decode($actual, true));
     }
 
+
+    public function testEncodeEntityWithToOneEntityExtension(): void
+    {
+        $productId = Uuid::randomHex();
+
+        $this->productRepository->create([
+            [
+                'id' => $productId,
+                'productNumber' => Uuid::randomHex(),
+                'stock' => 1,
+                'name' => 'Test product',
+                'price' => [['currencyId' => Defaults::CURRENCY, 'gross' => 10, 'net' => 8.10, 'linked' => false]],
+                'tax' => ['name' => 'test', 'taxRate' => 5],
+                'manufacturer' => [
+                    'id' => Uuid::randomHex(),
+                    'name' => 'shopware AG',
+                    'link' => 'https://shopware.com',
+                ],
+                'toOne' => [
+                    'name' => 'test',
+                ],
+            ],
+        ], Context::createDefaultContext());
+
+        $criteria = new Criteria([$productId]);
+        $criteria->addAssociation('toOne');
+
+        $productDefinition = $this->getContainer()->get(ProductDefinition::class);
+
+        /** @var ProductEntity $product */
+        $product = $this->productRepository->search($criteria, Context::createDefaultContext())->get($productId);
+        $encoder = $this->getContainer()->get(JsonApiEncoder::class);
+        $encodedResponse = $encoder->encode(new Criteria(), $productDefinition, $product, SerializationFixture::API_BASE_URL, SerializationFixture::API_VERSION);
```

Comment:
@AlexBachmann your tests seem to fail. The `encode()` method expects an array of metadata as 5th parameter, but you give an int. Can you have a look at that? https://github.com/shopware/platform/blob/832f7282064cb47c7e32a2d60a8d3f2bff9030a8/src/Core/Framework/Api/Serializer/JsonApiEncoder.php#L35

Reply:
That's right. The signature of the method changed since i started the pull request. I updated the tests so they should now pass.